### PR TITLE
Add unit tests for isDirectory, isFile and friends

### PR DESCRIPTION
**Pull request by tobsan**
This adds tests that check that this family of functions only return
true if the argument is of the expected type and false otherwise.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

### CODE REVIEW

**Review by breakreturn — APPROVED** [review]

File: common/unit-test/softwarecontainer-common_unittest.cpp [hunk 1]
```diff
@@ -57,3 +83,81 @@ TEST_F(SoftwareContainerCommonTest, ReturnCodeisSuccess) {
     ASSERT_TRUE(isSuccess(ReturnCode::FAILURE) == false);
 }
 
+/*
+ * Test that a directory is only seen as a directory and not as a file, socket etc.
+ */
+TEST_F(SoftwareContainerCommonTest, DirIsOnlyDir) {
+    std::string dirStr = getTempPath(true, false);
```
Comment (kursatkobya) [inline common/unit-test/softwarecontainer-common_unittest.cpp]: maybe using alias for readability would be useful 
`isDirectory = true`
`shouldUnlink = false`
`getTempPath(isDirectory, shouldUnlink);`

File: common/unit-test/softwarecontainer-common_unittest.cpp [hunk 2]
```diff
@@ -57,3 +83,81 @@ TEST_F(SoftwareContainerCommonTest, ReturnCodeisSuccess) {
...
+    ASSERT_FALSE(isSocket(dirStr));
+
+    ASSERT_EQ(rmdir(dirStr.c_str()), 0);
+}
+
+/*
+ * Test that a regular file is only recognized as a file and not as a dir, socket etc.
+ */
+TEST_F(SoftwareContainerCommonTest, FileIsOnlyFile) {
+    std::string fileStr = getTempPath(false, false);
```
Comment (kursatkobya) [inline common/unit-test/softwarecontainer-common_unittest.cpp]: maybe using alias for readability would be useful

File: common/unit-test/softwarecontainer-common_unittest.cpp [hunk 3]
```diff
@@ -57,3 +83,81 @@ TEST_F(SoftwareContainerCommonTest, ReturnCodeisSuccess) {
...
+
+    // Cleanup
+    ASSERT_EQ(unlink(fileStr.c_str()), 0);
+}
+
+/*
+ * Test that a socket is seen as a socket, but not as a file or dir etc
+ */
+TEST_F(SoftwareContainerCommonTest, SocketIsOnlySocket) {
+    std::string fileStr = getTempPath(false, true);
```
Comment (kursatkobya) [inline common/unit-test/softwarecontainer-common_unittest.cpp]: maybe using alias for readability would be useful

File: common/unit-test/softwarecontainer-common_unittest.cpp [hunk 4]
```diff
@@ -57,3 +83,81 @@ TEST_F(SoftwareContainerCommonTest, ReturnCodeisSuccess) {
...
+
+    // Cleanup
+    ASSERT_EQ(unlink(fileStr.c_str()), 0);
+}
+
+/*
+ * Test that a pipe is seen only as a pipe and not as a socket, file or dir.
+ */
+TEST_F(SoftwareContainerCommonTest, PipeIsOnlyPipe) {
+    std::string pipeStr = getTempPath(false, true);
```
Comment (kursatkobya) [inline common/unit-test/softwarecontainer-common_unittest.cpp]: maybe using alias for readability would be useful

**Review by kursatkobya — CHANGES_REQUESTED** [review]

File: common/unit-test/softwarecontainer-common_unittest.cpp [hunk 5]
```diff
@@ -47,7 +73,7 @@ TEST_F(SoftwareContainerCommonTest, bool2ReturnCode) {
     ASSERT_TRUE(bool2ReturnCode(false) == ReturnCode::FAILURE);
 }
 
-TEST_F(SoftwareContainerCommonTest, ReturnCodeisError) {
+TEST_F(SoftwareContainerCommonTest, ReturnCodzeisError) {
```
Comment (thenor) [inline common/unit-test/softwarecontainer-common_unittest.cpp]: Codze?

**Review by thenor — APPROVED** [review]
I am just wondering if there could be more corner cases.

File: common/unit-test/softwarecontainer-common_unittest.cpp (hunk 5 above)
Comment (tobsan) [reply common/unit-test/softwarecontainer-common_unittest.cpp]: Typo! Will fix!

File: common/unit-test/softwarecontainer-common_unittest.cpp (hunk 1 above)
Comment (tobsan) [reply common/unit-test/softwarecontainer-common_unittest.cpp]: Probably a good idea, yes.

**Review by kursatkobya — APPROVED** [review]

**Review by thenor — APPROVED** [review]
